src/snowflake/snowpark/modin/plugin/_internal/binary_op_utils.py

```diff
@@ -312,314 +300,527 @@ def _op_is_between_timedelta_and_numeric(
     )
 
 
-def compute_binary_op_between_snowpark_columns(
-    op: str,
-    first_operand: SnowparkColumn,
-    first_datatype: DataTypeGetter,
-    second_operand: SnowparkColumn,
-    second_datatype: DataTypeGetter,
-) -> SnowparkPandasColumn:
-    """
-    Compute pandas binary operation for two SnowparkColumns
-    Args:
-        op: pandas operation
-        first_operand: SnowparkColumn for lhs
-        first_datatype: Callable for Snowpark Datatype for lhs
-        second_operand: SnowparkColumn for rhs
-        second_datatype: Callable for Snowpark DateType for rhs
-        it is not needed.
+class BinaryOp:
+    def __init__(
+        self,
+        op: str,
+        first_operand: SnowparkColumn,
+        first_datatype: DataTypeGetter,
+        second_operand: SnowparkColumn,
+        second_datatype: DataTypeGetter,
+    ) -> None:
+        self.op = op
+        self.first_operand = first_operand
+        self.first_datatype = first_datatype
+        self.second_operand = second_operand
+        self.second_datatype = second_datatype
+        self.result_column = None
+        self.result_snowpark_pandas_type = None
+
+    @staticmethod
+    def is_binary_op_supported(op: str) -> bool:
+        """
+        check whether binary operation is mappable to Snowflake
+        Args
+            op: op as string
+
+        Returns:
+            True if binary operation can be mapped to Snowflake/Snowpark, else False
+        """
+
+        return op in SUPPORTED_BINARY_OPERATIONS
+
+    @staticmethod
+    def create(
+        op: str,
+        first_operand: SnowparkColumn,
+        first_datatype: DataTypeGetter,
+        second_operand: SnowparkColumn,
+        second_datatype: DataTypeGetter,
+    ) -> "BinaryOp":
+        def snake_to_camel(snake_str: str) -> str:
+            """Converts a snake case string to camel case."""
+            components = snake_str.split("_")
+            return "".join(x.title() for x in components)
+
+        def get_all_subclasses(cls: Type[BinaryOp]) -> Set[Type[BinaryOp]]:
+            """Gets all subclasses of a given class."""
+            subclasses = set()
+            to_check = [cls]
+
+            while to_check:
+                parent = to_check.pop()
+                for child in parent.__subclasses__():
+                    if child not in subclasses:
+                        subclasses.add(child)
+                        to_check.append(child)
+
+            return subclasses
```

Comment:
This part seems unnecessarily complex. Can we have a dict mapping from string operations like `add` to classes like `AddOp`?

Reply:
I can simplify it a little bit but I still prefer this programmatic way instead of listing all pairs.

Reply:
This version LGTM.